xarray/namedarray/core.py

```diff
@@ -847,5 +856,118 @@ def _to_dense(self) -> Self:
         else:
             raise TypeError("self.data is not a sparse array")
 
+    def transpose(
```

Comment:
Is this permute_dims?
https://data-apis.org/array-api/latest/API_specification/generated/array_api.permute_dims.html

Reply:
Done!